From a274b32c5a97c4b8ae0a67c136c20a13b166a333 Mon Sep 17 00:00:00 2001
From: Sainath Grandhi <sainath.grandhi@intel.com>
Date: Fri, 2 Nov 2018 16:21:22 -0700
Subject: [PATCH 1074/1552] SOS: Guest x2APIC support for SOS

This patch adds support to use x2APIC in SOS when ACRN exposes x2APIC
to guests and supports x2APIC MSR emulation.

On older versions of ACRN which does not expose x2APIC in cpuid,
this patch does not do anything.

Tracked-On: PKT-1592
Tracked-On: https://github.com/projectacrn/acrn-hypervisor/issues/1717
Signed-off-by: Sainath Grandhi <sainath.grandhi@intel.com>
---
 arch/x86/acrn/acrn.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/acrn/acrn.c b/arch/x86/acrn/acrn.c
index 8283673..6b01b27 100644
--- a/arch/x86/acrn/acrn.c
+++ b/arch/x86/acrn/acrn.c
@@ -56,8 +56,8 @@ static void acrn_pin_vcpu(int cpu)
 
 static bool acrn_x2apic_available(void)
 {
-	/* do not support x2apic */
-	return false;
+	/* ACRN supports x2apic emulation */
+	return true;
 }
 
 static void __init acrn_init_mem_mapping(void)
-- 
2.7.4

